components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementFacade.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.idp.mgt.dao;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.application.common.model.*;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants;
+import org.wso2.carbon.identity.core.model.ExpressionNode;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.idp.mgt.AuthenticatorEndpointConfigServerException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementClientException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementServerException;
+import org.wso2.carbon.idp.mgt.model.ConnectedAppsResult;
+import org.wso2.carbon.idp.mgt.util.UserDefinedAuthenticatorEndpointConfigManager;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class IdPManagementFacade {
+
+    private static IdPManagementDAO dao;
+    private final UserDefinedAuthenticatorEndpointConfigManager endpointConfigurationManager =
+            new UserDefinedAuthenticatorEndpointConfigManager();
+
+    public IdPManagementFacade(IdPManagementDAO dao) {
+        this.dao = dao;
+    }
+
+    public List<IdentityProvider> getIdPs(Connection dbConnection, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        List<IdentityProvider> idpList = dao.getIdPs(dbConnection, tenantId, tenantDomain);
+        for (IdentityProvider idp : idpList) {
+            populateEndpointConfig(idp, tenantDomain);
+        }
```

Comment:
Do we need to populate the end point here ? Is all authenticator information is expected in these listing cases ?